zerver/actions/message_send.py

```diff
@@ -204,6 +204,7 @@ def get_recipient_info(
         # of this function for different message types.
         assert stream_topic is not None
         user_ids_muting_topic = stream_topic.user_ids_with_visibility_policy(UserTopic.MUTED)
+        user_ids_unmuting_topic = stream_topic.user_ids_with_visibility_policy(UserTopic.UNMUTED)
```

Comment:
This adds one more database query in the message send code path. Can we replace this with a single query which returns something like map from `user_id` to `visibility_policy` value for a particular topic?

Reply:
Yeah, we should definitely do something like that. Might be cleaner to have the map be the other (visibility_policy=>user_id), but I'm not sure as I've not looked at how this is used.